src/packtPublishingFreeEbook.py

```diff
@@ -197,6 +197,7 @@ def get_product_download_urls(product_id):
                     else:
                         target_download_path = os.path.join(self.cfg.download_folder_path)
                     full_file_path = os.path.join(target_download_path, '{}.{}'.format(file_name, file_extention))
+                    temp_file_path = os.path.join(target_download_path, 'download.tmp')
```

Comment:
There is a `tempfile` package in standard library - maybe we could download to temporary directory first, and than move to expected download path? It's just an idea.